docs/AddingNewProjects.md

```diff
@@ -0,0 +1,51 @@
+# Adding New Projects To The Repo
+
+Sample PR of final result: https://github.com/dotnet/aspnetcore/pull/41945
+
+## Creating a new project
+1. Create a new folder that will house your `.csproj` and other project-related files.
+2. (EXTREMELY IMPORTANT) Inside this new folder, make a new folder for the source files of your project. For regular functionality-adding project this will be called `src`. However, if you are adding a different kind of project, it will be called something more applicable (ex. `test/` for a test project).
+3. In VS, Add a new `Solution Folder` in the new folder from the first step.
+4. Create the project via the VS `Add` menu (select the folder -> right click -> Add -> follow the wizard).
+
+    **Note:** (Only applicable to `src/` projects) Depending on what kind of project you are creating, VS will create different files in your project. You might also want to add the following files:
+    - `PublicAPI.Shipped.txt`
+      - Lists publically visible APIs that are exported from your final compiled `.dll`.
+      - This only lists APIs that have already been shipped to customers and cannot be changed.
+      - There is an empty template at `eng/PublicAPI.empty.txt` for your reference. You can copy and rename the file to add it to your project. Make sure the name is exactly as shown above.
+    - `PublicAPI.UnShipped.txt`
+      - Lists publicly visible APIs that are exported from your final compiled `.dll`. If this is not configured properly, you will get build errors. VS will warn you though with green squiggly lines. If you see these squiggly lines, open the VS Quick Actions (CTRL + '.') and select the option to and it to the public API.
+      - This only lists APIs that have NOT already been shipped to customers. So, these can still change.
+      - There is an empty template at `eng/PublicAPI.empty.txt` for your reference. You can copy and rename the file to add it to your project. Make sure the name is exactly as shown above.
+    - `AssemblyInfo.cs`
+      - Lists various properties of your compiled assembly, such as to which other packages your `internal` properties and methods are available to.
+      - You can also expose internals via `@(InternalsVisibleTo)` items in your project file instead of using this `AssemblyInfo.cs` file.
+
+## Adding to the rest of the repo
+1. VS should have already registered your `.csproj` in the corresponding solution ([`.sln`](https://github.com/dotnet/aspnetcore/blob/586ccc8c895862b65645c4b0f979db1eecd29626/AspNetCore.sln)) and solution filter ([`.slnf`](https://github.com/dotnet/aspnetcore/blob/586ccc8c895862b65645c4b0f979db1eecd29626/src/Middleware/Middleware.slnf#L107-L109)) files.
+  - If VS has not already modified these files, make sure to add it manually as is visible in the example listed above.
```

Comment:
If you did create a project outside of `AspNetCore.sln` adding it manually with a plain text editor is nearly impossible. This should mention that adding an "Existing Project..." in VS

```suggestion
  - If VS has not already modified these files, open the `.slnf` you want to add the project to. Create a solution folder for your project it if doesn't exist already. Then right click solution foulder -> Add -> Existing Project... -> follow the wizard.
```